### ISSUE TITLE
wallpaper folders cannot be changed

### ISSUE THREAD

**Issue by Heliotrop** [GabMus/HydraPaper#7]
I could remove those which were originally present - because they pointed nowhere. But now I'm stuck to add new ones
![bildschirmfoto von 2018-01-25 07-36-52](https://user-images.githubusercontent.com/35147227/35374303-97e8824c-01a2-11e8-870f-febad4fe3974.png)

Any hint how to add this manually (config file?)

**Comment by Heliotrop** [comment]
**syslog shows:**
Jan 25 07:40:06 pirol nautilus[5548]: g_key_file_load_from_file: assertion 'file != NULL' failed
Jan 25 07:40:06 pirol nautilus[5548]: Could not establish a connection to Tracker: Failed to load SPARQL backend: Die Schlüsselwertedatei enthält nicht die Gruppe »DomainOntology«
Jan 25 07:40:15 pirol gnome-shell[1711]: origin
Jan 25 07:44:18 pirol systemd[1632]: flatpak-org.gabmus.hydrapaper-5848.scope: Failed to add PIDs to scope's control group: Permission denied
Jan 25 07:44:18 pirol systemd[1632]: Failed to start flatpak-org.gabmus.hydrapaper-5848.scope.
Jan 25 07:44:18 pirol systemd[1632]: flatpak-org.gabmus.hydrapaper-5848.scope: Unit entered failed state.
Jan 25 07:44:19 pirol gnome-system-mo[5769]: Allocating size to gnome-system-monitor 0x55b212dd22a0 without calling gtk_widget_get_preferred_width/height(). How does the code know the size to allocate?
Jan 25 07:44:19 pirol dbus[975]: [system] Activating via systemd: service name='org.freedesktop.hostname1' unit='dbus-org.freedesktop.hostname1.service'
Jan 25 07:44:19 pirol systemd[1]: Starting Hostname Service...
Jan 25 07:44:20 pirol dbus[975]: [system] Successfully activated service 'org.freedesktop.hostname1'
Jan 25 07:44:20 pirol systemd[1]: Started Hostname Service.

**Comment by evfool** [comment]
I just did build from source, and I still can not add folders, so it might not be a flatpak only issue.

**Comment by evfool** [comment]
FYI: The config file to add some more folders should be ~/.config/hydrapaper

**Comment by GabMus** [comment]
That's very strange. You can add new folders using this button

![image](https://user-images.githubusercontent.com/1358527/35385358-5426059a-01c8-11e8-9d73-2ebdb0579031.png)

If you already tried, did it work? If not, what happened when you pressed that button?

Also, please both @evfool and @Heliotrop tell me what distribution and desktop environment you're running.

**Comment by GabMus** [comment]
BTW, the config file is a simple json and the complete path is `~/.config/hydrapaper.json`

**Comment by evfool** [comment]
@GabMus : changing the config file manually did solve it, the tool is great, thanks for doing it, I've been manually creating triple-monitor wallpapers until now.
Regarding my environment: I am on Fedora 27 (wayland, gnome shell, etc), installed HydraPaper both from flatpak and was running it from source. I saw the dropdown, but clicking it did nothing.
I just realized that it works (opens the combo) if I open the GtkPopover (wallpapers folder), change focus to another window, then click the down arrow to open the dropdown, but if clicking the dropdown arrow after opening the popover without switching focus to another window it does not open the filepicker. I hope this helps.

**Comment by GabMus** [comment]
@evfool great to know that it works now, and that it works with a triple monitor setup! I didn't test that since I only have 2, so thanks for that!

As for this switching focus bug, that's really weird if you ask me. I didn't experience it on my Arch system. Maybe (but can't be sure of it) this problem is wayland related? Try running on xorg and report back if it works or not.

**Comment by evfool** [comment]
Indeed, it works if running with GDK_BACKEND=x11 flatpak run org.gabmus.hydrapaper, so it surely is Wayland. Probably a simple testcase will also reproduce it, I guess it deserves a report for GTK.

**Comment by GabMus** [comment]
Yeah, totally sounds like a Gtk issue. Will try to report that

**Comment by GabMus** [comment]
Bug opened: https://bugzilla.gnome.org/show_bug.cgi?id=792895

**Comment by Heliotrop** [comment]
Yes, tried to use the button, but nothing happens. Sorry, but can you show an example config syntax?

**Comment by Heliotrop** [comment]
... and yes, I'm using wayland
json looks like:
`{"monitors": {"DEL - DVI-D-1": null, "GSM - DVI-I-1": null}, "selection_mode": "double", "wallpapers_paths": []}`

**Comment by evfool** [comment]
@Heliotrop: try clicking the Wallpapers folder, focus another window, and click the dropdown afterwards, it did the trick for me. Regarding the config file, just add whatever folders you want to the wallpapers_path, e.g. "wallpapers_paths": ["/home/username/Pictures","/usr/share/backgrounds/gnome"] is the default, I think

**Comment by GabMus** [comment]
> "wallpapers_paths": ["/home/username/Pictures","/usr/share/backgrounds/gnome"] is the default, I think

yes, it is.

@Heliotrop If you want to re-add the default folders in an easier way, just run `rm ~/.config/hydrapaper.json`. This will re-generate a new config file with default values on startup.

**Comment by Heliotrop** [comment]
Thanks, yes, putting it manually worked for me (Ubuntu 17.10). Great tool!

**Comment by GabMus** [comment]
Awesome. I'm sorry you're getting this issue with the file chooser, it's apparently a gtk bug and I can't do much about it unfortunately.

**Comment by Heliotrop** [comment]
I understand. It works like a charm now. Selecting and Applying is working since the path is correct.

Thanks again for your help

PS.: There's still a couple of bugs with wayland, since Ubuntu switched there from Unity. Sometimes Nautilus freezes completely ... but this is an entire different story.

**Comment by GabMus** [comment]
@Heliotrop You can still fallback to the xorg session if you want to.

**Comment by GabMus** [comment]
In 26b5a7ba2fd7c1baf05cdf753c59f27d32627674 I did a design change that removes the combobox/filechooserbutton in favor of a simple button. This should fix the problem on wayland. Please test it and report back.

[hydrapaper-1.1.3-82-gc8917d1.zip](https://github.com/GabMus/HydraPaper/files/1724065/hydrapaper-1.1.3-82-gc8917d1.zip)
